### PR TITLE
Increase maximum line length in source code formatting style from 80 to 120

### PR DESCRIPTION
**Pull request by clinssen**
This is a follow-up to #1176.

The purpose of this PR is to change the static code checker rules, such that the maximum line length within *.cpp, *.h and *.py source code files is increased from 80 to 120. All *.cpp and *.h files are then reformatted with clang-format, which causes the corresponding patch (48ea8e6) to be quite large. Reviewers might like to start with the other commits in this PR first.

@jessica-mitchell: are there any places in the developer documentation that should be updated with this information?

### CODE REVIEW

**Comment by jakobj** [comment]
while I welcome the general spirit of this change, just one small request: please give all PRs prior to this one a chance to be merged before merging this one. we've already had many, many annoying merge issues updating various branches when the 80 character limit was introduced.

**Comment by clinssen** [comment]
A live discussion about this subject will be scheduled for the Open NEST Developer Conference of April 29th.

@jakobj: although I welcome the spirit of your request, given the age of many open PRs, waiting for them all to be merged before merging this one is not feasible. Note that reformatting of all C++ files (.h, .cpp) can be done in a fully automated fashion.

**Comment by jakobj** [comment]
@clinssen though agreeing with your point that one can not wait for all PRs to be merged, I would like to ask that the obviously active PRs should get a chance to be merged since at least one of the maintainers (could it be me?) still has not managed to get `clang-format-3.6` running to automate the reformatting and will hence have to fight with merge issues ;)

**Comment by Silmathoron** [comment]
if this is adopted (hopefully not for the python files), I guess it would be a good idea to move formatting from the antediluvian clang-format-3.6 to a more recent version (e.g 6, as discussed in #1152) in the same PR, to get all the annoying stuff done in one go ;)

**Comment by clinssen** [comment]
@Silmathoron : in response to your comment on clang-format versioning, I was recently faced with the same issue. I ended up copying-and-pasting the following lines from the Travis build script:

https://github.com/nest/nest-simulator/blob/4c1e1f6bbe2d0a894057def7d0fdb76cbf26c802/extras/travis_build.sh#L133

This made the installation process quite painless. We could consider bundling the binary into our own git submodule in nest/nest-simulator, depending on the projected longevity of the llvm.org link. In addition I would recommend adding a note on this in the installation documentation. @jessica-mitchell : do you have any thoughts on what the best place for this note should be? Could we add a "Developer Tools" subheader under User Documentation → Installation Instructions?

**Comment by jessica-mitchell** [comment]
@clinssen Yeah, that sounds like a good idea. As of right now, we have the developer space ghpages for dev-related topics. So I would think the best thing to do would be to add something to the developer space and link to it from the installation page.

**Comment by Silmathoron** [comment]
@clinssen thanks, I just checked it and it seems to work also on non ubuntu-based plateforms, so it looks like a plan :+1:

File: extras/static_code_analysis.sh [hunk 1]
```diff
@@ -53,6 +53,7 @@ IGNORE_MSG_PEP8=${16}         # true or false, indicating whether PEP8 messages
 PEP8_IGNORES="E121,E123,E126,E226,E24,E704"
 PEP8_IGNORES_EXAMPLES="${PEP8_IGNORES},E402"
 PEP8_IGNORES_TOPO_MANUAL="${PEP8_IGNORES_EXAMPLES},E265"
+PEP8_MAX_LINE_LENGTH=120
```
Comment (gtrensch) [inline extras/static_code_analysis.sh]:
```suggestion

# PEP8 rules.
PEP8_MAX_LINE_LENGTH=120
```

**Comment by clinssen** [comment]
@jarsi noted that comments are not reformatted (e.g. `// line 1`, `// line 2` into `// line 1 line 2`) . In the future, we could consider setting `ReflowComments` to `true` in `.clang-format`. Leaving it at `false` is the more conservative option and we suggest to stick with `false` for this PR.

**Comment by clinssen** [comment]
@steffengraber noted that function declarations in some files (e.g. `block_vector.h`) are formatted according to `IndentWrappedFunctionNames=true`, even though in our `.clang-format`, it is set to `false`. Possibly, these indentations were added by hand. We could consider setting `IndentWrappedFunctionNames=true` in the future.

**Pull request merged**